packages/react-testing/src/mount.ts

```diff
@@ -127,6 +133,14 @@ export function createMount<
       resolveRoot: (root) => root.find(element.type),
     });
 
+    if (cleanup) {
+      const originalDestroy = wrapper.destroy.bind(wrapper);
+      wrapper.destroy = () => {
+        cleanup(wrapper, options);
```

Comment:
Would it be a lot of effort to support async `cleanup`? A common pattern in web is "loading state tests need end with a `graphQL.resolveAll()` to complete pending `React.act` calls".

If we could automate that, adding an `afterEach` check for dangling acts becomes less noisy.

Reply:
My worry is that `destroy` is currently sync, which means that existing uses could break if its unexpectedly changed, and simply overwriting it with an async function in some cases doesn't fix types. I can take a crack at it though.

Reply:
also @GoodForOneFare  if you have any ideas about how to deal with the points above (or think they're not a big deal) that's also good info :P

Reply:
Those are good points! I figured it wouldn't be trivial.

Overnight, I talked myself into wanting an alternative solution. Something more explicit, and part of general test flow.

The bad idea version of this looks like:

```ts
await component.trigger('onFocus', {andTotesResolveGraphQLForMe: true});
```